Assets/Scripts/NodeGameObjectFactory.cs

```diff
@@ -1,16 +1,62 @@
-﻿using UnityEngine;
+﻿using Assets.Scripts.Common;
+using Assets.Scripts.Model;
+using UnityEngine;
 
 namespace Assets.Scripts
 {
   public class NodeGameObjectFactory : MonoBehaviour
   {
+    TextureCache _materialCache;
+
     public GameObject nodePrefab;
     public GameObject ghostNodePrefab;
+    public GameObject nodeCanvasPrefab;
+
+    const PrimitiveType c_defaultNodeType = PrimitiveType.Sphere;
+
+    public void Start()
+    {
+      _materialCache = FindObjectOfType<TextureCache>();
+    }
+
+
+    public GameObject CreateNodeGameObject(Node node)
+    {
+      if (_materialCache == null)
+        _materialCache = FindObjectOfType<TextureCache>();
+
+      if (node.nodeClass == null)
+        return CreateDefaultNode(node.Position, Quaternion.Euler(node.Rotation));
+
+      var material = _materialCache.GetByClassId(node.nodeClass.id);
+
+      var gameObject = node.nodeClass.shape == null
+        ? GameObject.CreatePrimitive(c_defaultNodeType)
+        : GameObject.CreatePrimitive(node.nodeClass.shape.Value);
+
+      gameObject.transform.localScale = Vector3.one * (node.nodeClass.scale ?? 1);
+      gameObject.transform.position = node.Position;
```

Comment:
is this better than Instantiate for position and rotation?

Reply:
What do you mean?
The problem here is that Instantiate is for prefabs, and since we do not know shape and texture of the created object, there can be no prefab for it. Hence the custom logic to create the object from the bottom up

Reply:
ok, i thought it is for GameObject